### PR TITLE
Fix non-mac build with clang

### PR DESCRIPTION
**Pull request by klokik**
Unix implementation of networkInterfaceHelper is not compiled if
building with clang compiler on linux. This adds clang to lookup
list.

### CODE REVIEW

**Comment by christophe-calmejane** [comment]
Nice catch, thanks!
Although, I think it would probably be better to remove any reference to compiler ID here, as we want to compile a different file based on target host (not sure we want to handle cross-compilation right now) not based on compiler.

**Comment by christophe-calmejane** [comment]
Well, actually, we can use *CMAKE_SYSTEM_NAME* as it's defined by default to *CMAKE_HOST_SYSTEM_NAME*, and it will be correctly defined when cross-compiling.
I propose to remplace all the *if* with that, and add an *else* condition with a *message(FATAL_ERROR)*

**Comment by klokik** [comment]
Updated. There still might be other not fully correct target platform checks that rely on host system, e.g. implying that only MSVC is the only valid compiler for WIN32.  But that will need a thorough check one by one.

File: CMakeLists.txt [hunk 1]
```diff
@@ -204,7 +204,7 @@ endif()
 
 ############ Compiler compatibility
 
-if(WIN32)
```
Comment (christophe-calmejane) [inline CMakeLists.txt:207]: This change is actually incorrect.
Currently, I only want to support MSVC on windows (either microsoft clang implementation or VC++). As microsoft-clang is actually just a different toolset, it's still MSVC that is defined, so I still want to check for the version and exit if it's too old.

**Comment by christophe-calmejane** [comment]
I'll cherry-pick your 2 commits (not the last one), they won't be available right now as I'm working on a private clone at the moment, but it will be shortly!

Thanks a lot

**Comment by christophe-calmejane** [comment]
One question though, why the pattern matching for linux.
Isn't a strict equality enough (with "Linux")?
What is CMAKE_SYSTEM_NAME returning on your system?
CMake documentation says it should return Linux.

**Comment by klokik** [comment]
It does return "Linux", but since the implementation is not strictly linux specific, one may add say "(Linux)|(FreeBSD)|(QNX)" later, when it is needed, tested, and ironed out.

**Comment by klokik** [comment]
Well, I see there are linux specific includes, and weird ioctl's to segregate interface type. Still, seems portable enough.

**Pull request closed without merging**